### ISSUE TITLE
Big Sur updating guide

### ISSUE THREAD

**Issue by davidAlaBer** [zearp/OptiHack#27]
Hi @zearp!
Big sur is now in the App Store ready to download. I already installed it in my iMac and I like it. it would be great if you can have a time and guide us and write some steps to upgrade safely on the Optiplex or any considerations about it.
Best,
@davidAlaBer
----------------
"This guide has been tested with macOS Catalina and Big Sur but should work older version too. For the time being please stick with Catalina as Big Sur is still in beta testing."

**Comment by zearp** [comment]
Given your EFI up to date to the repo you can just install Big Sur or upgrade to it.

**Comment by zearp** [comment]
Given your EFI is up to date with the repo you can just install Big Sur or upgrade to it.

**Comment by davidAlaBer** [comment]
Thanks! I will update soon!

Enviado desde mi iPhone

**Comment by zearp** [comment]
Keep us posted and be sure to make a backup before updating!

**Comment by mgrimace** [comment]
FYI, updated to Big Sur and lost 4k (back to default resolution 2560x1600) - display shows 'iMac' vs., 'Retina iMac' in the Display settings as well (previously it would show Retina iMac when 4k was working).

**Comment by davidAlaBer** [comment]
Big Sur update from Catalina went fine! SwitchResX installed and 2560x1440 60hz Hidpi. It looks fine. I don't understand your point.

**Comment by mgrimace** [comment]
> Big Sur update from Catalina went fine! SwitchResX installed and 2560x1440 60hz Hidpi. It looks fine. I don't understand your point.

The point I'm raising is that my monitor is 4k which is a resolution of 3840x2160, my maximum current resolution in Big Sur is 2560x1600. This was an issue that we problem-solved successfully in Mojave, but it looks like the issue might have returned.

**Comment by davidAlaBer** [comment]
My monitor is also 4k and I can see 3840x2160 in Sys preferences>displays.
In SwitchResX I can see up to 7680x4320

**Comment by mgrimace** [comment]
Any changes to the default EFI poster here? Strange. I'm using DisplayPort, had 4k prior on Mojave no problem (I contributed the framebuffer data settings for the 4k fix originally). Using a 27" 4k display

**Comment by davidAlaBer** [comment]
9020 MT, dedicated Graphic Card rx 570, 27" 4k display, DisplayPort. Please feel free if I can give you any further info.

**Comment by mgrimace** [comment]
Thank you for the additional details! I now see the disconnect, I'm speaking specifically to using the iGPU (HD4600) for driving 4k monitors, which was a problem were able to solve on the previous version of MacOS, but seems to be giving me some difficulty again. A dGPU would definitely help, but is less viable on the small form factor (SFF) models due to size/power constraints

**Comment by zearp** [comment]
@mgrimace Just did some quick digging, it appears WhateverGreen'` ```enable-hdmi20``` option is currently not working with Big Sur. That would explain it. Gotta wait for a fix cuz I don't think there's a way to enable hdmi20 with a search/replace but I could be wrong.

@davidAlaBer I'm happy to hear the update went smoothly!

**Comment by zearp** [comment]
@mgrimace Just did some quick digging, it appears WhateverGreen's ```enable-hdmi20``` option is currently not working with Big Sur. That would explain it. Gotta wait for a fix cuz I don't think there's a way to enable hdmi20 with a search/replace but I could be wrong.

@davidAlaBer I'm happy to hear the update went smoothly!

**Comment by zearp** [comment]
@mgrimace 👀 

It is of course a bit confusing that we need a "HDMI 2.0" patch to make 4k work properly when using DisplayPort to DisplayPort cables. How that works is above my pay grade. I do know some cables can convert DP <-> HDMI signals, some both ways some only one way (found that out the hard way haha). I still can't test it myself cuz I spent my pocket money on a NUC instead of a 4k screen, but it's on my wish list for sure!

**Comment by zearp** [comment]
https://github.com/acidanthera/bugtracker/issues/1024
https://github.com/acidanthera/bugtracker/issues/1093
https://github.com/acidanthera/bugtracker/issues/1237

**Comment by mgrimace** [comment]
> Gotta wait for a fix cuz I don't think there's a way to enable hdmi20 with a search/replace but I could be wrong.

I tried the old hacky find/replace patch, confirmed not working. I'll wait for an update from WEG! 

> It is of course a bit confusing that we need a "HDMI 2.0" patch to make 4k work properly when using DisplayPort to DisplayPort cables. How that works is above my pay grade. 

Right? I took a look at the bug trackers and it seems like it's Lilu (user patching) that's generally broken altogether (and WEG relies on Lilu). I guess it'll depend on if/when a fully functioning Lilu is released for this to work properly. Might have to downgrade in the meantime for my poor eyes!

**Comment by zearp** [comment]
Yea we have to wait for fixes. I had posted some bug tracker urls here but somehow it also creates notifications on those bug reports cuz I mentioned it here so I removed them again. Didn't realise if we link to a bug here it will spam the linked thread with a notice that we posted it.

I will stick to Catalina for my eyes, I can't get used to the iPadOS feel of Big Sur. Doesn't feel like a professional Desktop OS to me compared to Catalina. but that's of course very subjective, many people love the new spacious UI elements. I don't 😅

**Comment by zearp** [comment]
ea we have to wait for fixes. I had posted some bug tracker urls here but somehow it also creates notifications on those bug reports cuz I mentioned it here so I removed them again. Didn't realise if we link to a bug here it will spam the linked thread with a notice that we posted it.

I will stick to Catalina for my eyes, I can't get used to the iPadOS feel of Big Sur. Doesn't feel like a professional desktop OS to me compared to Catalina, but that's very subjective, many people love the new spacious UI elements. I just don't 😅

**Comment by zearp** [comment]
@mgrimace Have you tried applying the patch directly to the kext? It’s a dirty method but I think it may work.

**Comment by mgrimace** [comment]
> @mgrimace Have you tried applying the patch directly to the kext? It’s a dirty method but I think it may work.

That's interesting, I have absolutely no idea where to begin doing that though! I can try it out, do you mean find/replacing the values manually in the AppleIntelFramebufferAzul kext?

For anyone curious I'll paste the old patch below, it went under root/kernal/patch in your OC config.plist. 

Key | Type | Value
-- | -- | --
Base | String |  
Comment | String | Enable 4k
Count | Number | 0
Enabled | Boolean | True
Find | Data | 0300220D 00030303 00000002 00003001
Identifier | String | com.apple.driver.AppleIntelFramebufferAzul
Limit | Number | 0
Mask | Data |  
Replace | Data | 0300220D 00030303 00000006 00000003
ReplaceMask | Data |  
Skip | Number | 0

**Comment by zearp** [comment]
Yes it's exactly that, very easy with a [hex editor](https://ridiculousfish.com/hexfiend/). The "tricky" part is making that modified kext load. You'll need to fully disable SIP including authenticated root. This part is easy by setting ```csr-active-config``` to ```FF0F0000``` in the config. Verify in recovery if they have been applied properly with ```csrutil status``` + ```csrutil authenticated-root status```. If not you may have to clear NVRAM and/or manually disable them with csrutil.

To save myself some typing, you can use info provided [here](https://medium.com/@andv/making-wifi-on-big-sur-unsupported-macs-with-failed-with-66-error-36c98e3f7965). Making the snapshots in that guide is optional. It's easy to reverse without making snapshots. Worst case you have to boot with a Big Sur installer and remove the kext and rerun kmutil.

As for patching the kext itself, there's a binary file inside it and it's really just a matter of finding the string and replacing it. Of course this kind of patching shouldn't be done anymore but for now this is the only way to apply these patches.

**Comment by mgrimace** [comment]
I appreciate the information here! I don't think I'll use it, but it great to learn the *_how_*. If there's no progress on Lilu/WEG I might give this a shot down the road but I think I'm in the same place you are - no objective need or subjective interest in Big Sur yet.

**Comment by mgrimace** [comment]
Interesting, following some discussion on Reddit here's how another person (on an Acer) was able to get 4k working on Big Sur with the HD4600. Note that they removed the device-id entirely, omitted some of the patches we have, and swapped one of the connector types from DP to HDMI. Still clarifying some points, but I'll post this here if it helps while I experiment a bit:

`<dict>
    <key>AAPL,ig-platform-id</key>
    <data>AwAiDQ==</data>
    <key>framebuffer-fbmem</key>
    <data>AAAAAw==</data>
    <key>framebuffer-patch-enable</key>
    <data>AQAAAA==</data>
    <key>framebuffer-stolenmem</key>
    <data>AAAABA==</data>
    <key>framebuffer-con1-enable</key>
    <data>AQAAAA==</data>
    <key>framebuffer-con1-type</key>
    <data>AAgAAA==</data>
</dict>`

**Comment by mgrimace** [comment]
Here's their settings in non-code (table) which enabled 4k on their HD4600. 

Key | Type | Value
-- | -- | --
AAPL,ig-platform-id | Data |  0300220D
framebuffer-fbmem | Data | 00000003
framebuffer-patch-enable | Data | 01000000
framebuffer-stolenmem | Data | 00000004
framebuffer-con1-enable | Data | 01000000
framebuffer-con1-type | Data | 00080000

They said they started with the basics (lines 1-4 which are the same frame buffer settings as ours with the removal of some entries), then they just played with the connector settings until it worked, and setting the con as HDMI enable 4k/30.

I went back and find our original connections which were con0 and con3 (which we've removed from the .plist since updating the DeviceProperties successfully for 4k). We had originally defined them as:

Key | Type | Value
-- | -- | --
framebuffer-con0-alldata | Data | FF000900 00000000 87000000
framebuffer-con0-enable | Data | 01000000
framebuffer-con3-alldata | Data | FF000000 00000000 40000000
framebuffer-con3-enable | Data | 01000000

It looks like the connector settings in the [Dortania guide](https://dortania.github.io/OpenCore-Post-Install/gpu-patching/intel-patching/connector.html) suggests patching it as:

framebuffer-patch-enable = 01000000
Enables WhateverGreen's patching mechanism
framebuffer-conX-enable = 01000000
Enables WhateverGreen's patching on conX
framebuffer-conX-type = 00080000
Sets the port to HDMI(<00 08 00 00>)
**Note:** framebuffer-conX-type= 0004000 would set the port as DP.

**Test 1:** Staying on Catalina, I'm going to try the removal of the other entries to match above, and re-includ our ports con0 and con3 (but set as DP not HDMI) and see if that keeps 4k.

**Comment by mgrimace** [comment]
1. Test 1: remove everything but platform-id and framebuffer settings, define connectors con0 and con3 as DP - **Fail** (boots; max resolution 1440p)
2. Test 2: same, define con0 and con3 as HDMI - **Fail**
3. Test 3: Define back as con0 and con3 as DP, then add other patches back one-by-one to see what enables 4k

**minimum result that enables 4k**

Key | Type | Value
-- | -- | --
AAPL,ig-platform-id | Data | 0300220D
framebuffer-fbmem | Data | 00000003
framebuffer-patch-enable | Data | 01000000
framebuffer-stolenmem | Data | 00000004
framebuffer-con0-enable | Data | 01000000
framebuffer-con0-type | Data | 00040000
framebuffer-con3-enable | Data | 01000000
framebuffer-con3-type | Data | 00040000
enable-hdmi20 | Data | 01000000

**Summary** on Catalina, the values we need for 4k is the frame buffer settings, and the _enable-hdmi20_ patch, that currently isn't working on Big Sur. There may be a way to get Big Sur working following above, but as it is there's no universal solution that I can find.

**Comment by zearp** [comment]
@mgrimace Good sleuthing there!

I'm not sure about the connector types, I always assumed it was best to match those up to the hardware and only change the types if its absolutely necessary. We have 2x DisplayPort out on these machines and they should be mapped as such, or don't use any mapping at all. I didn't really experiment much but I recall things for me (as simple dual 1080p monitor) user worked with and without those connector entries. But it's been a while.

Spoofing the device id is most likely not needed but I'll have to test that. I kind of stopped tinkering with this setup cuz everything works so well and stable for me. But I'll take another dive into this soon.

I'm still not sure if changing or even setting up the connector types would do anything, but if it helps 4k and doesn't break anything then there's no problem of course. My guess is that things will work with the correct framebuffer options as you figured out last time. Things might start working  once Lilu/WEG have fixed those issues without any changes

Keep me posted on your research and testing and thank you for contributing!

**Comment by zearp** [comment]
@mgrimace Good sleuthing there!

I'm not sure about the connector types, I always assumed it was best to match those up to the hardware and only change the types if its absolutely necessary. We have 2x DisplayPort out on these machines and they should be mapped as such, or don't use any mapping at all. I didn't really experiment much but I recall things for me (as simple dual 1080p monitor user) worked with and without those connector entries. But it's been a while.

Spoofing the device id is most likely not needed but I'll have to test that. I kind of stopped tinkering with this setup cuz everything works so well and stable for me. But I'll take another dive into this soon.

I'm still not sure if changing or even setting up the connector types would do anything, but if it helps 4k and doesn't break anything then there's no problem of course. My guess is that things will work with the correct framebuffer options as you figured out last time. Things might start working once Lilu/WEG have fixed those issues without any changes

Keep me posted on your research and testing and thank you for contributing!

**Comment by mgrimace** [comment]
> Spoofing the device id is most likely not needed but I'll have to test that.

Removing _device-id_, _disable-external-gpu_, _disable-hdmi-patches_, and _hda-gfx_ didn't seem to cause any immediate/obvious problems for me. However, there's also no obvious downside in including them either (and might be better to include since I don't have an eGPU, etc.). 

The only patches that were critical for 4k on Catalina that I found were the platform-id, framebuffers (fbmem, stolenmem), and _enable-hdmi20_. 

> My guess is that things will work with the correct framebuffer options as you figured out last time. Things might start working once Lilu/WEG have fixed those issues without any changes

Agreed! I was hopeful to find settings that would work on Big Sur without breaking Catalina. Yeah, I think it's best just to wait for Lilu/WEG and keep what we've got since it's working well/stable. Fun to tinker though :)

**Comment by zearp** [comment]
Have you tried applying the Catalina patches directly to the Big Sur kext?

I think there are 2 ways to try that, either patching the kext directly or patch it with OpenCore. I'm using the latter to [whitelist](https://github.com/zearp/Nucintosh/blob/master/EFI/OC/config.plist#L453-L467) an Apple Airport card that doesn't work on the SMBIOS I use for my NUCs.

Those cards can be found super cheap but for some reason it only works on certain models. The patch just replaces a supported model with the one I use. Pretty simple and effective. Something similar could work for the Azul kext.

**Comment by zearp** [comment]
ave you tried applying the Catalina patches directly to the Big Sur kext?

I think there are 2 ways to try that, either patching the kext directly or patch it with OpenCore. I'm using the latter to [whitelist](https://github.com/zearp/Nucintosh/blob/master/EFI/OC/config.plist#L453-L467) an Apple Airport card that doesn't work with the SMBIOS I use for my NUCs.

Those cards can be found super cheap but for some reason it only works on certain models. The patch just replaces a supported model with the one I use. Pretty simple and effective. Something similar could work for the Azul kext.

**Comment by zearp** [comment]
Have you tried applying the Catalina patches directly to the Big Sur kext?

I think there are 2 ways to try that, either patching the kext directly or patch it with OpenCore. I'm using the latter to [whitelist](https://github.com/zearp/Nucintosh/blob/master/EFI/OC/config.plist#L453-L467) an Apple Airport card that doesn't work with the SMBIOS I use for my NUCs.

Those cards can be found super cheap but for some reason only work on certain models. The patch just replaces a supported model with the one I use. Pretty simple and effective. Something similar could work for the Azul kext.

**Comment by mgrimace** [comment]
> Have you tried applying the Catalina patches directly to the Big Sur kext?

Not yet! I'm still on Catalina, no rush to Big Sur personally - mostly, it came up on reddit that 4k was possible, and I thought I'd see if I could update/optimize our DeviceProperties to enable 4k on both Catalina and Big Sur. Mostly a curiosity thing. As it is, I haven't yet found a way around the _enable-hdmi20_ requirement, but I figured I'd post my efforts in case anyone has any ideas!

> Edit: In case you need it, the identifier to use in OpenCore for the Azul kext is com.apple.driver.AppleIntelFramebufferAzul.

Thanks! This looks like the way to go if I do upgrade before Lilu/WEG are updated(or if they can't fix the issue). Basically, hot patch it via OpenCore directly like our old method.

Thanks and all the best!

**Comment by mgrimace** [comment]
Ok, update and potential fix re. _enable-hdmi20_, 4k on Big Sur, etc.

From one of the WEG devs/contributors: "For Intel IGPUs, another approach is to patch the framebuffer driver to raise the pixel clock limit (instead of patching CoreDisplay.framework to skip pixel clock validation). I added support for this to the latest version of [WhateverGreen](https://github.com/acidanthera/WhateverGreen/blob/18d991c9cd5c731546b30d0ed11d9b2e5da99a3d/Manual/FAQ.IntelHD.en.md#hdmi-in-uhd-resolution-with-60fps) (take a look at the HDMI in UHD resolution with 60FPS section); this also works on Big Sur (as it doesn't depend on the userspace patcher). Note that this isn't present in the latest release of WhateverGreen, so you'll have to build it from source or use the CI artifacts (visit https://github.com/acidanthera/WhateverGreen/actions, select the latest commit, and select Artifacts to download the latest binaries)." 

The enable HDMI flag seems more related to raising the max pixel clock limit from 450 MHz (insufficient for 4k/60) to 675 MHz, which might account for why we needed it (even though we're not technically using HDMI). It looks like with a recent update to WEG (not yet released), that we can accomplish this by adding the _enable-max-pixel-clock-override_ property to _IGPU_ or use the _-igfxmpc_ boot-arg to instead to raise the max pixel clock limit without relying on a userspace patch (which aren't working on Big Sur). 

I'll try and test this over the holidays!

TL;DR it's possible that the _-igfxmpc_ boot-arg can replace/fix the broken _enable-hdmi20_ to enable 4k on Big Sur.

**Comment by zearp** [comment]
@mgrimace Nice finds, so we have to sort of overlock in order to get high resolution output. I'm still stuck in 1080p land where myself so I can't contribute much to this. Thank you for your time and effort!

**Comment by zearp** [comment]
@mgrimace Nice finds, so we have to sort of overlock in order to get high resolution output. I'm still stuck in 1080p land myself so I can't contribute much to this. Thank you for your time and effort!

**Comment by mgrimace** [comment]
No problem! Unfortunately, so far none of my tests have panned out. Summary of findings:

- _enable-hmdi20_ is needed for 4k on Catalina, even though we don't use HDMI
- _enable-hmdi20_  is a userspace patch that is not working on Big Sur. 
- Its function is to skip validation of the pixel clock to enable HDMI 2.0 (which in turn enables 4k/60).
- The default pixel clock value is 450 MHz which isn't sufficient for 4k/60. There are a few patches that raise the max pixel clock limit to 675 MHz (which enables 4k/60)
- **Trial 1:** Replacing _enable-hdmi20_ with  _max-pixel-clock-frequency_ (set to enabled) **loses** 4k on Catalina
- **Trial 2:** Removing _enable-hdmi20_ and adding the boot-arg _-igfxmpc_ also **loses** 4k on Catalina

I know there's ways to get 4k working on Big Sur via patching, but my goal was to try and figure out a way that works on both Catalina and Big Sur without patching. Unfortunately, I haven't been able to crack this (theoretical) optimal DeviceProperties yet!

**Comment by fabiannagel** [comment]
Sharing my experiences on Big Sur. Currently stuck on WQHD.
Grabbed the most recent WhatEvergreen kext and tried using boot-arg `-igfxmpc`, no success.

I'm not sure how to properly use `max-pixel-clock-frequency` to go up to 675 MHz, would it be like this? 
```
<key>enable-max-pixel-clock-override</key>
				<data>01000000</data>
				<key>max-pixel-clock-frequency</key>
				<data>675000000</data>
```

@mgrimace The patches you've mentioned that successfully raise the pixel clock limit, are they Catalina-only? Would you share them so I can look into them? Thanks!

**Comment by mgrimace** [comment]
Hello @fabiannagel and thank you for testing this out!

The ones you're testing were the only two patches I was able to find. I'm pulling most of my info re. clock frequencies from the HDMI in UHD section here: https://github.com/acidanthera/WhateverGreen/blob/18d991c9cd5c731546b30d0ed11d9b2e5da99a3d/Manual/FAQ.IntelHD.en.md#hdmi-in-uhd-resolution-with-60fps

As far as I understand you do **not** need to set the clock frequency. My understanding is that adding the following to DeviceProperties:

```
<key>enable-max-pixel-clock-override</key>
<data>01000000</data>
```

OR,

`-igfxmpc` in the boot args section

Either would automatically raise the clock speed from 450Mhz to 675 Mhz (which is the default override value, I think.). You would likely want to delete:

`enable-hdmi20`

The other thing to try would be to build your own version of WEG. The updates that reflect these patches haven't made it into the latest release yet (as far as I know).

The steps to build your own:

visit https://github.com/acidanthera/WhateverGreen/actions, select the latest commit, and select Artifacts to download the latest binaries.

It'll download as a zip file, unzip (I chose the debug one but I'm not sure). Then replace WEG with the newest one you downloaded to your OC/Kexts folder. After you add the new values and the new WEG, I believe you also then have to do an OC snapshot in the menu of ProperTree with your config.plist.

Let me know if any of that is unclear and appreciate your testing!
~M

**Comment by fabiannagel** [comment]
Yeah, my last post was already with the WhateverGreen's latest version from source, as you described. Sadly, neither `enable-max-pixel-clock-override` on its own nor the combination with `max-pixel-clock-frequency` (675000000) has any effect. But you're right, judging from [this commit](https://github.com/acidanthera/WhateverGreen/commit/9b68d9fba30a08f72dfbfe4cf82f674039f531ac#diff-9052a6934eea7704d350f3ac9396d12f69220ae155973768cdc7521109488114), inputting the actual frequency should not be necessary.

As WEG's file name does not change with the bleeding edge version, I didn't do any snapshotting. Correct me if it does something else I'm not aware of.

Maybe `-igfxmpc`'s author @al3xtjames can shed some light on this?

**Comment by al3xtjames** [comment]
You can use `ioreg -r -k startup-timing | awk '/IOFBTimingRange/ {print $4}' | tr -d '<>'` to print the IOTimingRange property. This uses the [`IODisplayTimingRange` structure](https://github.com/acidanthera/MacKernelSDK/blob/06325f9ae288e16fb9838cf641ab18cd52ff934e/Headers/IOKit/graphics/IOGraphicsTypes.h#L697); the maxPixelClock field is at offset 0x28 (40). You should see `C0 AE 3B 28 00 00 00 00` (little-endian 0x283BAEC0, which is 675000000 in decimal) if the patch was successful.

**Comment by fabiannagel** [comment]
Thanks a lot for your help @al3xtjames. I am indeed seeing `c0ae3b2800000000` at output position 80 with `-igfxmpc` as boot arg. So the patch seems to work, but I'm still stuck at WQHD. Are there any additional steps that worked for you?

Also, I noticed in Hackintool (Patch/Info/Selected Framebuffer) that a previous try of mine to spoof a similar Broadwell GPU via `device-id` has persisted somehow: The selected framebuffer is stuck at an Iris Plus 655. Trying to enforce my actual HD4600 results in a fallback to software rendering (at least in 4K, sigh). So at this moment I'm unsure if an unrelated issue on my machine might prevent your patch from properly allowing 4K output. But it should work on Catalina as well, right?

**Comment by mgrimace** [comment]
Here is a response I received on reddit, this user has 4k/60 working on a mobile version of HD4600, I haven't had a chance to look at the values (there's definitely some ones I haven't seen before) but with a bit of tweaking it might help:

`<key>AAPL,ig-platform-id</key>
<data>CAAuCg==</data>
<key>AAPL,slot-name</key>
<string>Internal@0,2,0</string>
<key>device-id</key>
<data>EgQAAA==</data>
<key>device_type</key>
<string>VGA compatible controller</string>
<key>disable-external-gpu</key>
<data>AQAAAA==</data>
<key>dpcd-max-link-rate</key>
<data>FAAAAA==</data>
<key>enable-backlight-registers-fix</key>
<data>AQAAAA==</data>
<key>enable-dpcd-max-link-rate-fix</key>
<data>AQAAAA==</data>
<key>enable-lspcon-support</key>
<data>AQAAAA==</data>
<key>enable-max-pixel-clock-override</key>
<data>AQAAAA==</data>
<key>force-online</key>
<data>AQAAAA==</data>
<key>force-online-framebuffers</key>
<data>AAAAAAAAAAM=</data>
<key>framebuffer-con0-alldata</key>
<data>AAAIAAIAAAAwAAAAAQUJAAAEAAAHAQAAAgQKAAAEAAAHAQAA</data>
<key>framebuffer-con0-enable</key>
<integer>1</integer>
<key>framebuffer-cursormem</key>
<data>AAAAAw==</data>
<key>framebuffer-fbmem</key>
<data>AAAAAw==</data>
<key>framebuffer-patch-enable</key>
<integer>1</integer>
<key>framebuffer-portcount</key>
<data>BAAAAA==</data>
<key>framebuffer-stolenmem</key>
<data>AAAACg==</data>
<key>framebuffer-unifiedmem</key>
<data>AAAAgA==</data>
<key>hda-gfx</key>
<string>onboard-1</string>
<key>model</key>
<string>Intel HD Graphics 4600</string>`

**Comment by mgrimace** [comment]
My copy/paste above is a mess, here's the values in an easier to read table format. Note I've made no changes from the original poster who had these values for a mobile version of the 4600 - as in, please don't copy/paste/use these directly. I'm pasting here because it looks like there's a lot of leads here to track down the necessary properties for Big Sur (and ideally also working on Catalina).

Key | Type | Value
-- | -- | --
AAPL,ig-platform-id | Data | 08002E0A 
AAPL,slot-name | String | Internal@0,2,0
device-id | Data | 12040000
device_type | String | VGA compatible controller
disable-external-gpu | Data | 01000000
dpcd-max-link-rate  | Data | 14000000
enable-backlight-registers-fix | Data | 01000000
enable-lspcon-support | Data | 01000000
enable-max-pixel-clock-override | Data | 01000000
force-online | Data | 01000000
force-online-framebuffers | Data | 01000000
framebuffer-con0-alldata | Data | 00000800 02000000 30000000 01050900 00040000 07010000 02040A00 00040000 07010000
framebuffer-con-enable | Number | 1
framebuffer-fbmem | Data | 00000003
framebuffer-patch-enable | Number | 1
framebuffer-portcount | Data | 04000000
framebuffer-stolenmem | Data | 00000004
framebuffer-unifiedmem | Data | 00000080
hda-gfx | String | onboard-1
model | String | Intel HD Graphics 4600

**Comment by tohusebo** [comment]
Any update on this @mgrimace ? I am trying to get a Hd 5500 working with my internal 4k display on a XPS 9343. I can`t get hardware acceleration working.

**Comment by mgrimace** [comment]
hi @tohusebo I've tracked my progress in this pull request: https://github.com/zearp/OptiHack/pull/33

I closed the pull, but I was able to get 4k working without user-space patches but could not get dual monitor support. Basically 4k or dual monitor but not both simultaneously. I believe it's related to properly defining the connectors but I couldn't work that out on my own (hit the limit of my understanding). Feel free to check out the pull and if you have any ideas I'm definitely open to suggestions.

Zearp posted some other tips in the discussion above to manually patch if needed. My intention with this trial was to get the right properties to get 4k working stable on both Catalina and Big Sur without needing to patch anything (i.e., to optimize the DeviceProperties for our specific build.).  Worth a shot and made some progress, but can't get further on my own!
